tests/test_mg_convert/test_mg_convert.py

```diff
@@ -0,0 +1,206 @@
+                    mg_cross_sections_file = \
+                        mg_cross_sections_file.convert_representation(
+                            conv[0], conv[1], conv[1])
+        elif convert[0] in ['legendre', 'tabular', 'histogram']:
+            mg_cross_sections_file = \
+                mg_cross_sections_file.convert_scatter_format(
+                    convert[0], convert[1])
+        elif convert[0] in ['angle', 'isotropic']:
+            mg_cross_sections_file = \
+                mg_cross_sections_file.convert_representation(
+                    convert[0], convert[1], convert[1])
+
+    mg_cross_sections_file.export_to_hdf5()
+
+
+class MGXSTestHarness(PyAPITestHarness):
+    def _build_inputs(self):
+        # Instantiate some Macroscopic Data
+        uo2_data = openmc.Macroscopic('UO2')
+
+        # Instantiate some Materials and register the appropriate objects
+        mat = openmc.Material(material_id=1, name='UO2 fuel')
+        mat.set_density('macro', 1.0)
+        mat.add_macroscopic(uo2_data)
+
+        # Instantiate a Materials collection and export to XML
+        materials_file = openmc.Materials([mat])
+        materials_file.cross_sections = "./mgxs.h5"
+        materials_file.export_to_xml()
+
+        # Instantiate ZCylinder surfaces
+        left = openmc.XPlane(surface_id=4, x0=-5., name='left')
+        right = openmc.XPlane(surface_id=5, x0=5., name='right')
+        bottom = openmc.YPlane(surface_id=6, y0=-5., name='bottom')
+        top = openmc.YPlane(surface_id=7, y0=5., name='top')
+
+        left.boundary_type = 'reflective'
+        right.boundary_type = 'vacuum'
+        top.boundary_type = 'reflective'
+        bottom.boundary_type = 'reflective'
+
+        # Instantiate Cells
+        fuel = openmc.Cell(cell_id=1, name='cell 1')
+
+        # Use surface half-spaces to define regions
+        fuel.region = +left & -right & +bottom & -top
+
+        # Register Materials with Cells
+        fuel.fill = mat
+
+        # Instantiate Universe
+        root = openmc.Universe(universe_id=0, name='root universe')
+
+        # Register Cells with Universe
+        root.add_cells([fuel])
+
+        # Instantiate a Geometry, register the root Universe, and export to XML
+        geometry = openmc.Geometry(root)
+        geometry.export_to_xml()
+
+        settings_file = openmc.Settings()
+        settings_file.energy_mode = "multi-group"
+        settings_file.batches = batches
+        settings_file.inactive = inactive
+        settings_file.particles = particles
+
+        # Create an initial uniform spatial source distribution
+        bounds = [-5, -5, -5, 5, 5, 5]
+        uniform_dist = openmc.stats.Box(bounds[:3], bounds[3:])
+        settings_file.source = openmc.source.Source(space=uniform_dist)
+
+        settings_file.export_to_xml()
+
+    def _run_openmc(self):
+        # Run multiple conversions to compare results
+        cases = [['legendre', 2], ['legendre', 0],
```

Comment:
I really like the way you designed this test to parametrically explore all of the conversion pathways - it's too bad we don't have more spare time on our Travis build! I wouldn't be opposed to commenting out a subset of the options below to relax the runtime requirements for this test.